### PR TITLE
Device selection

### PR DESCRIPTION
**Pull request by grunskis**
Shows list of configured components after the onboarding and allows to add new components. There's no API to get devices by type so currently you can only add new Philips Hue components (there's a hardcoded deviceId)

### CODE REVIEW

File: senic_hub/setup_app/components/DeviceSelection.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    this.state = {
+      component: props.navigation.state.params,
+    }
+  }
+
+  render() {
+    return (
+      <List>
+        <FlatList
+          data={this.state.component.selected_devices}
```
Comment (larsblumberg) [inline senic_hub/setup_app/components/DeviceSelection.js]: Is `selected_devices` coming from our backend's API or why do we use an underscore here?

File: senic_hub/setup_app/components/SelectComponentDevices.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        'Content-Type': 'application/json',
+      },
+      body: body,
+    }
+    console.log('sending body', body, 'params', params)
+    fetch(API_URL + '/-/nuimos/0/components', params)
+                   .then((response) => {
+                     console.log('response', response)
+                     navigation.navigate('NuimoComponents')
+                   })
```
Comment (larsblumberg) [inline senic_hub/setup_app/components/SelectComponentDevices.js]: This indentation got a little too much attention here ;)

File: senic_hub/setup_app/components/SetupDevices.js [hunk 3]
```diff
@@ -63,27 +67,30 @@ export default class SetupDevices extends Component {
     this.pollDevices()
   }
 
-  componentWillUnmount() {
+  clearTimeouts() {
```
Comment (larsblumberg) [inline senic_hub/setup_app/components/SetupDevices.js]: Why are timeouts not cleared any longer in `componentWillUnmount`? I.e. why was this change necessary?

Comment (grunskis) [inline senic_hub/setup_app/components/SetupDevices.js]: The screen is no longer unmountented when it's part of the navigation. Not sure why they don't do that... Keeping in the `componentWillUnmount` would hurt though. What do you think?

File: senic_hub/setup_app/components/SelectComponentDevices.js (hunk 2 above)
Comment (grunskis) [inline senic_hub/setup_app/components/SelectComponentDevices.js]: Yep. Still struggling to teach Emacs how to understand ES6+JSX... Will update...

File: senic_hub/setup_app/components/DeviceSelection.js (hunk 1 above)
Comment (grunskis) [inline senic_hub/setup_app/components/DeviceSelection.js]: Yep, coming from the API

File: senic_hub/setup_app/components/SelectComponentDevices.js (hunk 2 above)
Comment (grunskis) [inline senic_hub/setup_app/components/SelectComponentDevices.js]: Fixed in cf28853

File: senic_hub/setup_app/components/SetupDevices.js (hunk 3 above)
Comment (larsblumberg) [inline senic_hub/setup_app/components/SetupDevices.js]: For the records: We decided to address that with a different PR, using the WIX navigation component you suggested
